examples/config.py

```diff
@@ -0,0 +1,138 @@
+#   the documentation and/or other materials provided with the
+#   distribution.
+# * The name of the author may not be used to endorse or promote
+#   products derived from this software without specific prior
+#   written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+"""This script allows a user to launch a chain of NFs from a JSON
+config file passed in as an argument when running the script"""
+
+import sys
+import json
+import os
+import shlex
+import time
+from signal import signal, SIGINT
+from subprocess import Popen, PIPE
+
+def handler(signal_received, frame):
+    """Handles cleanup on user shutdown"""
+    print("\nExiting...")
+    for proc in procs_list:
+        try:
+            proc.kill()
+        except OSError:
+            pass
+    sys.exit(0)
+
+def get_config():
+    """Gets path to JSON config file"""
+    if len(sys.argv) < 2:
+        print("Error: Specify config file")
+        sys.exit(0)
+
+    file_name = sys.argv[1]
+    if os.path.exists(file_name):
+        return file_name
+    print("Error: No config file found")
+    sys.exit(0)
+
+def remove_prefix(text, prefix):
+    """Strip parameters in JSON config file"""
+    if text.startswith(prefix):
+        return text[len(prefix):]
+    return text
+
+def on_failure(error_output):
+    """Handles shutdown on error"""
+    print(error_output)
+    for proc in procs_list:
+        try:
+            proc.kill()
+        except OSError:
+            pass
+    for n in nf_list:
+        try:
+            os.system("sudo pkill " + n)
+        except OSError:
+            pass
+    print("Error occurred. Exiting...")
+    sys.exit(1)
+
+procs_list = []
+nf_list = []
+cmds_list = []
+if __name__ == '__main__':
+    if os.path.basename(os.getcwd()) != "examples":
+        print("Error: Run script from within /examples folder")
+        sys.exit(1)
+
+    signal(SIGINT, handler)
+
+    config_file = get_config()
+
+    with open(config_file) as f:
+        data = json.load(f)
+    for k, v in data.items():
+        for item in v:
+            nf_list.append(k)
+            PARAM = str(item).strip("'{[]}'")
+            PARAM = remove_prefix(PARAM, "u'parameters': u'")
```

Comment:
(goes along with my strip pylint comment). What's the PARAM argument doing? Can we do something like 
```py
cmds_list.append("./go.sh " v.parameter)
``` 
instead? to leverage the JSON syntax?

Reply:
So, I worked with doing something like v.parameter or item.parameter, but was having some issues dealing with encoding/decoding and then it not letting me concat unless item was a command. I just did some more research though, and I found a better way to do this. I'll push my edits in a few minutes!

Reply:
Awesome, great work!